torchao/quantization/qat/fake_quantizer.py

```diff
@@ -186,3 +189,32 @@ def __repr__(self) -> str:
         Return a human readable representation of this `FakeQuantizer` with config details.
         """
         return "FakeQuantizer(%s)" % self.config
+
+
+class _Float8ActivationFakeQuantizer(torch.nn.Module):
+    """
+    Simple fake quantizer for float8 fake quantization, intended for activations only.
+    """
+
+    FLOAT8_DTYPE = torch.float8_e4m3fn
+
+    def __init__(self, scaling_granularity: ScalingGranularity):
+        super().__init__()
+        self.enabled = True
+        self.scaling_granularity = scaling_granularity
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        if self.enabled:
+            original_dtype = x.dtype
+            scale = tensor_to_scale(
+                x,
+                self.FLOAT8_DTYPE,
+                scaling_granularity=self.scaling_granularity,
+                axiswise_dim=get_maybe_axiswise_dim(-1, self.scaling_granularity),
+            )
+            x_fq = x / scale
```

Comment:
this should be a multiply instead of a divide

source: https://github.com/pytorch/ao/blob/f0f1f6ce470d2854eb8fafc5dda3b4ffd80f1fe5/torchao/float8/float8_tensor.py#L153

you also should clamp before casting, source: https://github.com/pytorch/ao/blob/f0f1f6ce470d2854eb8fafc5dda3b4ffd80f1fe5/torchao/float8/float8_tensor.py#L154

Reply:
I'd recommend adding a unit test on this logic, you can check for SQNR >= ~20